src/mono/mono/.editorconfig

```diff
@@ -0,0 +1,21 @@
+[*]
+# We don't like trailing whitespace, but there are a lot of existing files that have it.  Don't automatically reformat them.
+trim_trailing_whitespace = false
+
+# See https://www.mono-project.com/community/contributing/coding-guidelines/
```

Comment:
Looks like this is about C#/.NET. Do we want to apply the same style to C as well, or is there a separate style guide for that?

Reply:
> Should we apply the formatting based on this editorconfig change to all the existing C files, or is this meant only for new files? If desired, [eclint](https://www.npmjs.com/package/eclint) could help do that.

We shouldn't apply the formatting, in my opnion.  The goal is not to churn the existing code, but to make it easier for contributors to make changes that follow the existing style.



> Looks like this is about C#/.NET. Do we want to apply the same style to C as well, or is there a separate style guide for that?

Historically Mono's C and C# code both followed the Mono Coding Guidelines.  However in the dotnet/runtime repo the C# code has generally followed the dotnet/runtime style.

For the C code we just need to capture the most important aspects, we shouldn't aim for completeness:
* tabs, not spaces; 8 column indent size
* brace placement
* spaces before open parens; spaces around operators.